src/python/CRABClient/JobType/Analysis.py

```diff
@@ -156,7 +156,7 @@ def run(self, filecacheurl = None):
                     msg += " When Data.lumiMask is not specified, Data.runRange can not include more than 50000 runs."
                     raise ConfigurationException(msg)
                 lumi_list = LumiList(runs = run_list)
-        if lumi_list:
+        if lumi_list != None:
```

Comment:
The key is in this line. If lumi_list is empty (but not None), we still want to pass to the server this information about empty lists of runs and lumis (currently we are not passing it, and the server runs over the full dataset). Or should the client already not submit to server?